server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesIndexSearcher.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.aggregations.timeseries;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.PriorityQueue;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.index.mapper.TimeSeriesIdFieldMapper;
+import org.elasticsearch.search.aggregations.BucketCollector;
+import org.elasticsearch.search.aggregations.LeafBucketCollector;
+
+import java.io.IOException;
+
+/**
+ * An IndexSearcher wrapper that executes the searches in time-series indices by traversing them by tsid and timestamp
+ * TODO: Convert it to use index sort instead of hard-coded tsid and timestamp values
```

Comment:
Oh, I don't click summit review, Now I start the review.

Reply:
I am not sure I understand your comment.

Reply:
Sorry, I describe  more clearly. Now It's in the time_series sense, the index is sort by _tsid and @timestamp, and the _tsid and  @timestamp must be a single value field, so it's ok.

But in a more common index sort sense, it may meet the multi-valued fields problem. e.g:
index is sort by foo, the setting is
```
index.sort.field = foo.
index.sort.order = asc.
index.sort.mode = min.
```

the sample doc:
```
doc1:  {"foo":[1,3,5,7]}
doc2:  {"foo":[2,4,6,8]}
doc3:  {"foo":[3,4,5]}
```

As `index.sort.mode = min`, To get the sorted doc, the order is doc1->doc2->doc3, but doc2\doc3 have some value that are small than doc1.
so traveling the `foo` field is not in a real sequential way.

The issue (https://github.com/elastic/elasticsearch/issues/80825) is doing to set the value to single-valued, when the sorted field is set to be a single-valued, it's ok to travel them sequential.